### PR TITLE
add weighted ppc

### PR DESCRIPTION
**Pull request by aloctavodia**
Allows to use `sample_ppc` to get  weighted samples, if more than one trace and a set of weights are passed.
The weights could come from the `compare` function, for example.

I will try to add a notebook with  examples ASAP.

### CODE REVIEW

File: pymc3/sampling.py [hunk 1]
```diff
@@ -536,6 +544,100 @@ def sample_ppc(trace, samples=None, model=None, vars=None, size=None,
     return {k: np.asarray(v) for k, v in ppc.items()}
 
 
+def sample_ppc_w(traces, samples=None, models=None, size=None, weights=None,
```
Comment (junpenglao) [inline pymc3/sampling.py]: IMO Your approach before was better - why adding a new function instead?

**Comment by aloctavodia** [comment]
I think is better for the user to have a separated function to perform a weighted ppc (model averaging).

File: pymc3/sampling.py (hunk 1 above)
Comment (aloctavodia) [reply pymc3/sampling.py]: I think it could be potentially confusing for the user.

**Comment by junpenglao** [comment]
I see, a notebook demonstrating how to use this for model averaging is necessary then. It would be quite a cool feature.

**Comment by aloctavodia** [comment]
I will try do add that notebook ASAP, thanks for  the review!

File: pymc3/sampling.py [hunk 2]
```diff
@@ -530,23 +530,94 @@ def sample_ppc(trace, samples=None, model=None, vars=None, size=None,
 
     seed(random_seed)
 
-    if weights is not None:
-        if len(trace) != len(weights):
-            raise ValueError(
-                'The number of traces and weights should be the same')
+    indices = randint(0, len(trace), samples)
+    if progressbar:
```
Comment (aseyboldt) [inline pymc3/sampling.py]: tqdm progress bars need to be closed if sampling is interrupted. Some try...finally would do it.

**Comment by aloctavodia** [comment]
I add an example showing how to use the new function and fix the tqdm issue
I also rename  the model comparison notebook and add it to the examples list, under the `Howto` category. I did it as part of this PR, because I reference that notebook in the new example.

**Comment by junpenglao** [comment]
Nice! I think it's ready to merge once the test pass.

**Comment by aseyboldt** [comment]
This looks nice! But I think we should merge only after the 3.1 release.

**Comment by junpenglao** [comment]
Fine with me.
@twiecki what is the time line for 3.1 release?

**Comment by twiecki** [comment]
@junpenglao After we fix the notebooks and examples we should be good to go.

**Comment by junpenglao** [comment]
@twiecki the examples are fixed (#2254) except one which depends on #2277. Since #2277 is on the 3.2 road map I think #2254 is OK to merge now.
As for the notebook, I think they are more or less ready as well (I fixed GLM_*, and @ferrine fixed VI notebooks).

**Comment by aloctavodia** [comment]
I just fix a couple of conflicts, that I think were introduced by #2291

**Comment by aloctavodia** [comment]
I just fix a couple of conflicts (probably in the less elegant way).

**Comment by fonnesbeck** [comment]
Thanks @aloctavodia

**Pull request merged**